pallets/capacity/src/lib.rs

```diff
@@ -767,6 +771,10 @@ impl<T: Config> Pallet<T> {
 		CapacityLedger::<T>::insert(target, capacity_details);
 	}
 
+	fn set_reward_pool(era: <T>::RewardEra, new_reward_pool: &RewardPoolInfo<BalanceOf<T>>) {
+		StakingRewardPool::<T>::set(era, Some(new_reward_pool.clone()));
+	}
```

Comment:
Should we mutate here ? set is cool assuming it serves dual purpose of remove/update but need to check

Reply:
I'm following our previous pattern of `let mut foo = Capacity::get_a_stored_thing(key)` followed by a set.  We've been moving away from mutate.